WcaOnRails/lib/results_validators/persons_validator.rb

```diff
@@ -126,13 +129,27 @@ def validate(competition_ids: [], model: Result, results: nil)
                                                LOWERCASE_NAME_WARNING,
                                                name: p.name)
           end
-          if split_name.length > 2
+          if split_name.any? { |n| n.upcase == n && n.length > 2 && n != 'DVM' }
+            @warnings << ValidationWarning.new(:persons, competition_id,
+                                               UPPERCASE_NAME_WARNING,
+                                               name: p.name)
+          end
+          if split_name.length == 1
+            @warnings << ValidationWarning.new(:persons, competition_id,
+                                               SINGLE_NAME_WARNING,
+                                               name: p.name)
+          elsif split_name.length > 2
             if split_name[1, split_name.length-2].any? { |n| n.length == 1 }
               @warnings << ValidationWarning.new(:persons, competition_id,
-                                                 MISSING_ABBREVIATION_PERIOD_WARNING,
+                                                 MISSING_PERIOD_WARNING,
                                                  name: p.name)
             end
           end
+          if split_name.any? { |n| n.chop.include? '.' }
```

Comment:
Just curious: What exactly does `.chop` do?

Reply:
It removes the last character of a string